hal/network/lwip/wiznet/wiznetif.cpp

```diff
@@ -419,16 +428,40 @@ int WizNetif::down() {
 }
 
 int WizNetif::powerUp() {
-    return 0;
+    // FIXME: As long as the interface is initialized,
+    // it must have been powered up as of right now.
+    if (pwrState_ != IF_POWER_STATE_UP) {
+        pwrState_ = IF_POWER_STATE_UP;
+        notifyPowerState();
+    }
+    return SYSTEM_ERROR_NONE;
 }
 
 int WizNetif::powerDown() {
-    return down();
+    int ret = down();
+    // FIXME: This don't really power off the module.
+    // Notify the system network manager that the module is powered down
+    // to bypass waitInterfaceOff() as required by system sleep.
+    if (pwrState_ != IF_POWER_STATE_DOWN) {
+        pwrState_ = IF_POWER_STATE_DOWN;
+        notifyPowerState();
+    }
+    return ret;
+}
+
+void WizNetif::notifyPowerState() {
```

Comment:
Let's add an argument here and task it with updating internal `pwrState_` as well. There shouldn't be any reason to do that externally and then still call into this method.